mesonbuild/environment.py

```diff
@@ -561,7 +579,8 @@ def __init__(self, source_dir, build_dir, options):
         user_options = collections.defaultdict(dict)  # type: T.DefaultDict[str, T.Dict[str, object]]
 
         # meson builtin options, as passed through cross or native files
-        meson_options = PerMachineDefaultable()  # type: PerMachineDefaultable[T.DefaultDict[str, T.Dict[str, object]]]
+        meson_options = PerMachineDefaultable(
+            collections.defaultdict(dict), collections.defaultdict(dict))  # type: PerMachineDefaultable[T.DefaultDict[str, T.Dict[str, object]]]
```

Comment:
More complicated structure again, no idea what that means... at least put a comment to tell what it contains.

Reply:
I'll add a comment to explain, but I've actually just moved the creation of the defaultdicts here, rather than being conditional.